### PR TITLE
Implement Transaction Receipt Handler

### LINKED ISSUE #12: Index Transaction Receipts
### Description

Currently, transaction receipts are not being indexed. Although, the receipts are not exposed by `near-api-js` ([source](https://github.com/near/near-api-js/blob/9189e118200f0300b28f32ffbc4f412fc63b9071/packages/near-api-js/src/providers/provider.ts#L85)), it can be fetched from the RPC call made [here](https://github.com/near/near-api-js/blob/9189e118200f0300b28f32ffbc4f412fc63b9071/packages/near-api-js/src/providers/json-rpc-provider.ts#L125). These receipts can be parsed and indexed alongside their corresponding transactions.

### PR DESCRIPTION
**Pull request by guplersaxanoid**
Fixes #12

### CODE REVIEW

File: packages/node/src/utils/near.ts [hunk 1]
```diff
@@ -80,31 +82,50 @@ export async function wrapBlock(
     const chunkResult = await api.chunk(chunk.chunk_hash);
     const transactionPromises = chunkResult.transactions.map(
       async (transaction) => {
-        const wrappedTx = await wrapTransaction(api, blockResult, transaction);
+        const exectuionOutcome = await api.txStatusReceipts(
```
Comment (stwiname) [inline packages/node/src/utils/near.ts:108]: Im not sure this is right. `txStatusReceipts` gets the receipts for a transactions but they may be over many blocks. So we can end up with data from other blocks.

Theres a couple of problems with this:
* We have data from future blocks
* This is non-deterministic when the project is up to date with the chain then the result of this query could change.


It seems that [chunk details](https://docs.near.org/api/rpc/block-chunk#chunk-details) contains the relevant receipts that we want for the block

File: packages/node/src/utils/near.ts [hunk 2]
```diff
@@ -80,31 +82,50 @@ export async function wrapBlock(
...
 }
 
-export async function wrapTransaction(
-  api: providers.JsonRpcProvider,
+export function wrapTransaction(
   block: BlockResult,
   txn: Transaction,
-): Promise<NearTransaction> {
-  const exectuionOutcome = await api.txStatusReceipts(txn.hash, txn.signer_id);
+  exectuionOutcome: any,
```
Comment (stwiname) [inline packages/node/src/utils/near.ts:154]: Type please

File: packages/node/src/utils/near.ts [hunk 3]
```diff
@@ -244,16 +299,63 @@ export function filterTransactions(
   );
 }
 
+export function filterReceipt(
```
Comment (stwiname) [inline packages/node/src/utils/near.ts:338]: We should have tests for this

**Review by stwiname — COMMENTED** [review]
It looks like we will need to update the dictionary and dictionary query to support receipts. Can you please do that as well. 

Ideally we get the dictionary running ASAP

File: packages/node/src/utils/near.ts [hunk 4]
```diff
@@ -244,12 +335,54 @@ export function filterTransactions(
...
+  receipt: NearTransactionReceipt,
+  filter?: NearReceiptFilter,
+): boolean {
+  if (!filter) return true;
+  if (filter.sender && receipt.predecessor_id !== filter.sender) return false;
```
Comment (stwiname) [inline packages/node/src/utils/near.ts:343]: Is the predecessor id the sender or the contract that emitted this receipt?

Comment (guplersaxanoid) [reply packages/node/src/utils/near.ts:343]: The account that emitted the receipt. signer id is the one that signed off the transaction

**Pull request merged**